tcms/testcases/views.py

```diff
@@ -763,7 +764,35 @@ def get(request, case_id):
         'test_case_run_status': TestCaseRunStatus.objects.all(),
         'bug_trackers': BugSystem.objects.all(),
     }
-    return render(request, 'case/get.html', context_data)
+
+    if test_plan:
+        url_params = "?from_plan=%d&case=%d" % (test_case.pk, test_plan.pk)
+    else:
+        url_params = "?case=%d" % test_case.pk
```

Comment:
I think this would look clearer written like this:
```python
url_params = "?case=%d" % test_case.pk
if test_plan:
        url_params += "&from_plan=%d" % test_case.pk
```
it also fits on fewer lines. Unless of course the order of the url params matters for some reason